src/js/lib/safe.js

```diff
@@ -159,26 +166,47 @@ Pebble.sendAppMessage = function(message, success, failure) {
 /* Wrap setTimeout and setInterval */
 var originalSetTimeout = setTimeout;
 window.setTimeout = function(callback, delay) {
+  if (safe.warnSetTimeoutNotFunction !== false && typeof callback !== 'function') {
```

Comment:
The default behavior for JavaScript is to silently fail and return a new timer ID. This confuses many developers using pebble.js, so I decided to put a one time warning that can be disabled if really necessary.